core/trino-main/src/main/java/io/trino/server/security/InsecureAuthenticator.java

```diff
@@ -53,11 +59,18 @@ public Identity authenticate(ContainerRequestContext request)
             user = basicAuthCredentials.get().getUser();
         }
         else {
-            user = emptyToNull(request.getHeaders().getFirst(PRESTO_USER));
+            try {
+                ProtocolHeaders protocolHeaders = detectProtocol(alternateHeaderName, request.getHeaders().keySet());
+                user = emptyToNull(request.getHeaders().getFirst(protocolHeaders.requestUser()));
+            }
+            catch (ProtocolDetectionException e) {
+                // ignored
+                user = null;
+            }
         }
 
         if (user == null) {
-            throw new AuthenticationException("Basic authentication or " + PRESTO_USER + " must be sent", BasicAuthCredentials.AUTHENTICATE_HEADER);
+            throw new AuthenticationException("Basic authentication or " + TRINO_HEADERS.requestUser() + " must be sent", BasicAuthCredentials.AUTHENTICATE_HEADER);
```

Comment:
This should use detected protocol header, as the code above accesses `protocolHeaders.requestUser()` header.

(or, we change the message not to quote header name at all)